src/main/java/core/basesyntax/serviceimpl/OperationStrategyImpl.java

```
@@ -0,0 +1,24 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.common.FruitTransaction;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlerMap) {
+        this.operationHandlerMap = operationHandlerMap;
+    }
+
+    @Override
+    public OperationHandler getOperationHandler(FruitTransaction.Operation operation) {
+        OperationHandler operationHandler = operationHandlerMap.get(operation);
+        if (operationHandler == null) {
+            throw new IllegalArgumentException("Unknown operation: " + operation);
```

getOperationHandler currently retrieves from the map and throws `IllegalArgumentException` when a handler is missing. The checklist requires informative RuntimeExceptions and validation: "Don't ignore exceptions in try-catch blocks ... rethrow a RuntimeException with an informative message and exception object." and the Implementation Details require validating the `operation` arg (non-null) and reporting missing handlers with informative exceptions. Change behavior to validate `operation` for null (throw informative RuntimeException) and, if the map lacks a handler, throw a RuntimeException with a clear message (include operation value). Do not rely on `IllegalArgumentException` here; use the project's convention of informative RuntimeException as required.